lumigator/jobs/inference/model_clients.py

```diff
@@ -76,60 +76,84 @@ def predict(
-        if self._pipeline.tokenizer.model_max_length != VERY_LARGE_INTEGER:
-            return
-        # Only override if it's the default value:
-        # i.e., VERY_LARGE_INTEGER for models that don't have a model_max_length explicity set
-        # Common parameter names to check in config
-        plausible_max_length_params = [
-            # BERT-based, LLaMA
-            "max_position_embeddings",
-            # GPT-2-based
-            "n_positions",
-            "n_ctx",
-            # ChatGLM-based
-            "max_sequence_length",
-            "seq_length",
-            # Mistral-based
-            "sliding_window",
-        ]
-
-        # Check config parameters
-        for param in plausible_max_length_params:
-            if hasattr(config, param):
-                value = getattr(config, param)
-                if isinstance(value, int) and value < VERY_LARGE_INTEGER:  # Sanity check for reasonable values
-                    self._pipeline.tokenizer.model_max_length = value
-                    logger.info(
-                        f"Setting the maximum length of input tokens to {value} \
-                                based on the config.{param} attribute."
-                    )
-                    return
-
-        # If no suitable parameter is found, warn the user and continue with the HF default
-        logger.warning(
-            f"Could not find a suitable parameter in the model config to set model_max_length. \
-                Using default value: {self._pipeline.tokenizer.model_max_length}"
+
+        # The reason I create the models and tokenizers separately is so that I can validate the model type
+        # before creating the pipeline. This way I can give a more helpful error message and stack trace
+        self._pipeline = pipeline(
+            task=hf_pipeline.task,
+            model=model,
+            tokenizer=self.tokenizer,
+            revision=hf_pipeline.revision,
+            device=hf_pipeline.device,
         )
+        self._set_max_length()
+        logger.info(f"HuggingFaceSeq2SeqPipeline initialized with config: {config}")
+
+    def _set_max_length(self):
+        """Make sure that the model can actually support the max_tokens configured.
+        For the Seq2Seq models, the max_length is the max_position_embeddings. That's because the input and output
+        tokens have separate positions, so the model can generate upto max_position_embeddings tokens.
+        This isn't true if it's a CausalLM model, since the output token positions would be len(input) + len(output).
+        """
+        max_length = self._pipeline.model.config.max_position_embeddings
+        # If the model is of the HF Hub the odds of this being wrong are low, but it's still good to check that the
+        # tokenizer model and the model have the same max_position_embeddings
+        if self._pipeline.tokenizer.model_max_length != max_length:
```

Comment:
Because some models (esp older ones) Don't have a tokenizer_config file! Looking at you, facebook/bart-large-cnn lol https://huggingface.co/facebook/bart-large-cnn/discussions/71